src/internal/pjs/api_server.go

```diff
@@ -255,6 +258,66 @@ func (a *apiServer) WalkJob(req *pjs.WalkJobRequest, srv pjs.API_WalkJobServer)
 	return nil
 }
 
+func (a *apiServer) Await(ctx context.Context, req *pjs.AwaitRequest) (*pjs.AwaitResponse, error) {
+	// handle job context and request validation.
+	var id pjsdb.JobID
+	if req.Context != "" {
+		jid, err := a.resolveJobCtx(ctx, req.Context)
+		if err != nil {
+			return nil, errors.Wrap(err, "resolve job ctx")
+		}
+		id = jid
+	} else { //nolint:staticcheck
+		// TODO(PJS): do auth.
+	}
+	// TODO(PJS): remove this once auth is implemented.
+	reqID := pjsdb.JobID(req.Job)
+	if id != reqID {
+		log.Error(ctx, "job context token does not match requested job.", zap.Int64("request.Job.ID", req.Job))
+		id = reqID // defaulting to this for testing until auth is implemented.
+	}
+
+	// poll database every second and cancel after 60 seconds
+	ctx, cancel := context.WithTimeout(ctx, DefaultRPCTimeout)
+	defer cancel()
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+				return nil, status.Errorf(codes.DeadlineExceeded, "Await timeout exceeded")
+			}
+			return nil, status.Errorf(codes.Canceled, "Await canceled: %v", ctx.Err())
+		case <-ticker.C:
+			var currentState pjs.JobState
+			if err := dbutil.WithTx(ctx, a.env.DB, func(ctx context.Context, sqlTx *pachsql.Tx) error {
+				job, err := pjsdb.GetJob(ctx, sqlTx, pjsdb.JobID(req.Job))
+				if err != nil {
+					if errors.As(err, &pjsdb.JobNotFoundError{}) {
+						return status.Errorf(codes.NotFound, "job %d not found", req.Job)
+					}
+					return errors.EnsureStack(err)
+				}
+				jobInfo, err := toJobInfo(job)
+				if err != nil {
+					return errors.Wrap(err, "inspect job")
```

Comment:
It would be best to wrap this with `toJobInfo(%d)` in case the implementation of toJobInfo changes.  if toJobInfo wants to wrap the code from InspectJob with `InspectJob`, that's fine.